api/src/main/java/org/openmrs/module/reporting/cohort/definition/library/BuiltInCohortDefinitionLibrary.java

```diff
@@ -103,6 +105,19 @@ public CohortDefinition getAnyEncounterDuringPeriod() {
         return new MappedParametersCohortDefinition(cd, "onOrAfter", "startDate", "onOrBefore", "endDate");
     }
 
+    @DocumentedDefinition("anyEncounterDuringPeriodWithOccurrence")
+    public CohortDefinition getAnyEncounterDuringPeriodWithOccurrence() {
+        EncounterCohortDefinition cd = new EncounterCohortDefinition();
+        cd.addParameter(new Parameter("onOrAfter", "reporting.parameter.startDate", Date.class));
+        cd.addParameter(new Parameter("onOrBefore", "reporting.parameter.endDate", Date.class));
+        cd.addParameter(new Parameter("atLeastCount", "reporting.parameter.atleast", Integer.class));
+        cd.addParameter(new Parameter("atMostCount", "reporting.parameter.atMost", Integer.class));
+        cd.addParameter(new Parameter("encounterTypeList", "reporting.parameter.encounterTypeList", EncounterType.class, List.class, null));
+        cd.addParameter(new Parameter("formList", "reporting.parameter.formList", Form.class, List.class, null));
+        cd.addParameter(new Parameter("locationList", "reporting.parameter.location", Location.class, List.class, null));
+        return new MappedParametersCohortDefinition(cd, "onOrAfter", "fromDate", "onOrBefore", "toDate", "encounterTypeList", "encounterTypes", "locationList", "location", "atLeastCount", "atLeast", "atMostCount", "atMost", "formList", "form");
```

Comment:
Alright, @mseaton that sounds good. However, I'll wait to see what @djazayeri has to say in this regard before refactoring.